### DISCUSSION TITLE
Vaillant heat pump and EEBUS

### DISCUSSION THREAD

**Discussion by rmalbrecht** [enbility/ship-go#16] [General]
Hi,
my Vaillant heatpump gained eeBus support via a sw-update some days ago. Now I'm curios what to do with that.
Is there a command line tool, ... to discover the services, see what is exposed, ....?
I can see the avahi announcement in my network, but that is currently all.
Regards,
Robert

**Comment by Kr0llx** [comment]
Hey @rmalbrecht you could use the [EEBus-Go-Lib](https://github.com/enbility/eebus-go/). The README shows you how to start a DEMO HEMS. Use this to connect to your heat pump. You can then find all exchanged messages in the terminal. If you need further help or have received the messages, please let us know.

Hope this helps.
There is also a Slack channel :+1:  https://enbility.net/contact/

**Comment by rmalbrecht** [comment]
I got it running (sort of).

go run cmd/main.go -remoteski 123456

seems to do something. There is a new connection showing up in the Vaillant app, where I have to confirm trusting the new device.
But then, nothing happens. No success or error message, no logs, no terminal output, ...
Is that intended behavior?

**Comment by Kr0llx** [comment]
Im also only a user of the lib so take everything with a grain of salt. Are you sure you are using the linked program. As the command should look more like the following:

`go run cmd/hems/main.go 4715`

This however generates a new Certificate at each start. Meaning you get a new SKI every time. Could you try storing the Cert and reusing it?
